src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,143 @@
+destroyed, which may be the right time to clean up any subscriptions that were registered during the
+connect process.
+
+### Scrolling over fixed size items
+If you're scrolling over a list of items that are all the same fixed size, you can use the
+`FixedSizeVirtualScrollStrategy`. This can be easily added to your viewport using the `itemSize`
+directive.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+The fixed size strategy also supports setting the buffer size, i.e. the number of items rendered
+beyond the edge of the viewport. This can be adjusted by setting the `bufferSize` input.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50" bufferSize="1">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+**Note: The fixed size strategy will likely be changed to allow specifying a separate
+`minBufferPx` and `addBufferPx` like the autosize strategy**
+
+The fixed size viewport strategy is less flexible than the autosize strategy because it requires all
+items to be the same size, but the advantage of this constraint is that it allows for better
+performance, since items do not need to be measured as they are rendered. 
+
+### Scrolling over items with different sizes
+If you're scrolling over a list of items with different sizes, you can use the
+`AutoSizeVirtualScrollStrategy`. This can be added to your viewport by using the `autosize`
+directive.
+
+```html
+<cdk-virtual-scroll-viewport autosize>
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+The `autosize` strategy allows the buffer to be configured through two inputs `minBufferPx` and
+`addBufferPx`. The `minBufferPx` is the minimum amount of buffer (in pixels) that the viewport
+should try to maintain on either side of the viewport. The `addBufferPx` is the amount of buffer
+(in pixels) that the viewport should try to render out when it detects that the buffer has dropped
+below the `minBufferPx`. It's helpful for this to be a little larger than the `minBufferPx` as it
+allows the viewport to render out new buffer items in batches rather than constantly needing to
+render new ones.
+
+```html
+<cdk-virtual-scroll-viewport autosize minBufferPx="50" addBufferPx="100">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+Because the auto size strategy needs to measure the size of the elements, its performance may not
+be as good as the fixed size strategy. 
+
+### Setting the viewport orientation
+The orientation of the viewport can be adjusted by using the `orientation` input. It defaults to
+`vertical` which virtualizes scrolling along the y-axis. It can be set to `horizontal` to virtualize
+scrolling along the x-axis. If you use this option you need to make sure that the content is
+actually laid out so that it extends along the x-axis. To do this you may want to target CSS at
+`.cdk-virtual-scroll-content-wrapper` which is the wrapper element that contains the rendered
```

Comment:
Would it be better/possible to recommend styling their own element instead of a class built into the virtual scroller?

Reply:
It's eventually going to be possible to specify that element themselves, or just allow it to be implicitly created if they don't specify. That's part of the work for making it work with `<table>`, `<ul>`, etc.